src/patch.jl

```diff
@@ -58,15 +58,19 @@ function apply!(pe::PatchEnv, patches)
 end
 
 function apply(body::Function, pe::PatchEnv)
-    original_pe = get_active_env()
+    prev_pe = get_active_env()
     set_active_env(pe)
     try
         return body()
     finally
-        set_active_env(original_pe)
+        set_active_env(prev_pe)
     end
 end
 
 function apply(body::Function, patches; debug::Bool=false)
     return apply(body, PatchEnv(patches, debug))
 end
+
+const PATCH_ENV = Ref{PatchEnv}(PatchEnv())
+set_active_env(pe::PatchEnv) = (global PATCH_ENV[] = pe)
```

Comment:
The global is not needed. We are changing content, not binding.

```suggestion
set_active_env(pe::PatchEnv) = (PATCH_ENV[] = pe)
```